### PR TITLE
Add ApplyBQSR, SamtoolStats, BamQC

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the [nf-core/sarek branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/sarek)
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Make sure your code lints (`nf-core lint .`).
- [ ] Documentation in `docs` is updated
- [ ] `CHANGELOG.md` is updated
- [ ] `README.md` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: modules/nf-core/software/samtools_index.nf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+    input:
+        tuple val(meta), path(bam)
+
+    output:
+        tuple val(meta), path(bam), path("*.bai")
+//    samtools index ${idSample}.bam
+
+    //    samtools index ${idSample}.recal.bam
+
```
Comment (MaxUlysse) [inline modules/nf-core/software/samtools_index.nf:14]:
```suggestion

```

File: modules/local/process/merge_bam.nf [hunk 2]
```diff
@@ -4,14 +4,16 @@ process MERGE_BAM {
     tag "${meta.id}"
 
     input:
-        tuple val(meta), path(bam), path(bai)
+        tuple val(meta), path(bam)//, path(bai) optional: true
```
Comment (MaxUlysse) [inline modules/local/process/merge_bam.nf:7]: `optional: true` for the input?

Comment (FriederikeHanssen) [reply modules/local/process/merge_bam.nf:7]: This is void now, sorry still working on this.

**Comment by FriederikeHanssen** [comment]
@MaxUlysse maybe don't merge quite yet. I have a hunch about the problem with BamQC, but am too tired now. Hopefully, I can sit down in the morning with it :)

**Comment by FriederikeHanssen** [comment]
Will merge, and we figure out the bamqc issues in the next PR

**Pull request merged**